InciCsvCorrector.java

```diff
@@ -0,0 +1,159 @@
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Class for the correction of the csv file containing the INCI ingredients found at
+ * "http://ec.europa.eu/growth/tools-databases/cosing/pdf/COSING_Ingredients-Fragrance%20Inventory_v2.csv".
+ * There are reoccurring formatting errors in the downloaded file that make it unusable (line endings inside the ingredient description),
+ * this class fixes those errors by writing in a new file the fixed text, the paths to the source and end files will be given by the user.
+ * The INCI database is updated quite often so this class can be used every time there is an update to get a usable file.
+ * @author Nicolò Cervo (g3)
+ */
+public class InciCsvCorrector {
+
+    private static final int DATE_LENGTH = 10;  // dd/mm/yyyy
+    private static final int MINIMUM_VALID_LENGTH = 15; // of a line
+
+    public static void main(String[] args)
+    {
+        String[] paths = getFilesPaths();
+        File source = new File(paths[0]);
+        File corrected = new File(paths[1]);
+        boolean needsCorrection = false;
+        ArrayList<Integer> invLines = null;
+
+        try {
+            invLines = findInvalidLines(source);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if(!(invLines.isEmpty())) {   // if there are invalid lines print their number
+            needsCorrection = true;
+            System.out.print("invalid lines : ");
+            while(!invLines.isEmpty()){
+                System.out.print(invLines.remove(0) + " ");
+            }
+            System.out.println();
+        }
+        if(needsCorrection){
+            try {
+                correctInterruptedLines(source, corrected);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            System.out.println("the file has been corrected");
+        }else{
+            System.out.println("the file does not need correction");
+        }
+    }
+
+    /**
+     * finds the invalid lines in the source file and prints them
+     * @param source source file
+     * @return an ArrayList of integers with the line numbers in the source file
+     * @throws IOException when the file does not exist
+     */
+    private static ArrayList<Integer> findInvalidLines(File source) throws IOException {
```

Comment:
You should throw a FileNotFoundException

Reply:
Here I was not sure what to do because I already catch the file not foun exception when i get te file so fileReader() should never trow an exception, i think.